pkg/services/interfaces.go

```diff
@@ -71,6 +71,9 @@ type NetworkProvider interface {
 	// HasLoadBalancer indicates whether this provider has a load balancer for Services.
 	HasLoadBalancer() bool
 
+	// SupportsVMRedinessProbe indicates whether this provider support vm readiness probe.
+	SupportsVMRedinessProbe() bool
```

Comment:
```suggestion
	SupportsVMReadinessProbe() bool
```

Reply:
Isn't this more a `RequiresVMReadinessProbe`?

Reply:
If we're using Avi as a Load Balancer provider, we do not require the VM readiness probe because Avi already has an in-built service endpoint health checker. However, NSX LoadBalancer doesn't share this feature. Hence, CAPV introduced the readiness probe.
The Kubernetes 1.19 update has an added feature, which allows retries for the kubeadm join operation. Ideally, if these retries can be successful before the timeout period (5 minutes), we can potentially eliminate the VM readiness check for all network provider types. However, we'd need to run a scale test before we entirely remove the VM readiness check for vsphere-vds and nsxt.
Concerning nsx-vpc, the VM readiness check would be unsuccessful, because the node network remains unreachable from the supervisor master node. That's the reason why I suggest we name the function to SupportsVMReadinessProbe().